MVJ-850 executor field

NC-jsAhonen opened the pull request:
Add executor to contracts and contract changes.

Related UI changes: https://github.com/City-of-Helsinki/mvj-ui/pull/611

NC-jsAhonen committed:
contract and contractchange: executor field

NC-jsAhonen committed:
executor: update serializer field fix

NC-jsAhonen committed:
contract_change executor

NC-jsAhonen committed:
sanitizerconfig changes

NC-jsAhonen committed:
Merge branch 'develop' into MVJ-850-executor-field

henrinie-nc commented on leasing/serializers/contract.py:
```diff
@@ -42,6 +46,13 @@ class ContractChangeCreateUpdateSerializer(
     )
+    executor = InstanceDictPrimaryKeyRelatedField(
+        instance_class=User,
+        queryset=User.objects.all(),
```
We should propably investigate if this can be filtered somehow, as we only want to show specific types of users.

NC-jsAhonen replied:
I created a function `get_users_for_executor_field` that seems to give the correct queryset, but the choices of the field are not yet filtered. This needs to be moved to a

henrinie-nc replied:
Implemented this check in another pr.

henrinie-nc commented on leasing/serializers/contract.py:
```diff
@@ -145,6 +159,13 @@ class ContractCreateUpdateSerializer(
     )
+    executor = InstanceDictPrimaryKeyRelatedField(
+        instance_class=User,
+        queryset=User.objects.all(),
```
https://github.com/City-of-Helsinki/mvj/pull/930/files#r2111397265

NC-jsAhonen committed:
queryset for executor field users

henrinie-nc committed:
add MvjUserManager and get_officers() to it, use in contract.executor

this manager method is likely to be used elsewhere in the app for getting only officer users

henrinie-nc committed:
add distinct() to get_officers(), join produces multiple

henrinie-nc commented on leasing/serializers/contract.py:
```diff
@@ -12,10 +16,37 @@
+    )
+
+    if not users.exists():
+        return User.objects.none()
```
this has been fixed in the other pr

henrinie-nc commented:
Moved this PR to https://github.com/City-of-Helsinki/mvj/pull/936
as I continued to work on it and wanted to clean the commit history from the merge.
